frmts/gtiff/gt_overview.cpp

```diff
@@ -368,6 +368,10 @@ CPLErr GTIFFBuildOverviewsEx(const char *pszFilename, int nBands,
                 nBandFormat = SAMPLEFORMAT_INT;
                 break;
 
+            case GDT_Float16:
+                CPLAssert(false);
```

Comment:
There should be a // TODO   /  // FIXME because at some point we will want the GTiff driver to support Float16.  Currently it does by transcoding from/to Float32

Reply:
That is now implemented via `GFloat16`.

Reply:
Done. I'm converting to Float32 and added a TODO.